deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/config/CompositeAuthenticationFactory.java

```diff
@@ -54,10 +52,7 @@ public ContainerRequestFilter create(TeletraanServiceContext context) throws Exc
                         .setAuthorizer(context.getAuthorizationFactory().create(context))
                         .buildAuthFilter();
 
-        List<AuthFilter> filters = new ArrayList<>();
-        filters.addAll(tokenFilters);
-        filters.add(envoyAuthFilter);
-
-        return new ChainedAuthFilter(filters);
+        return new ChainedAuthFilter(Arrays.asList(createScriptTokenAuthFilter(context),
+                createOauthTokenAuthFilter(context), envoyAuthFilter, createJwtTokenAuthFilter(context)));
```

Comment:
Yes it's working now. Evnoy changes are deployed. 

Functionally, the order mostly doesn't matter. It will matter in the following aspects
1. 99% requests currently use script token. To cut the unnecessary steps, I put it as the first. 
2. We have to enable authorization header pass through so both in-house Oauth and Envoy Oauth work during migration, thus some overlap there. This PR adjusts the order so a request isn't double authenticated. 
3. I supposes a test to ensure the order is helpful.

Reply:
I tried to add a test, but due to the class is generic, and it's provided by the dropwizard, it would involve a lot code changes to create the test. So I'll leave it as-is.